File: java/src/main/java/it/giannibombelli/wsc2026/booking/application/integration/payment/adapter/RefundRequest.java
package it.giannibombelli.wsc2026.booking.application.integration.payment.adapter;

import it.giannibombelli.wsc2026.booking.domain.events.BookingResultEvents;
import it.giannibombelli.wsc2026.common.utils.Require;
import it.giannibombelli.wsc2026.payment.integration.RefundRequestIntegrationCommand;

public final class RefundRequest {

    private RefundRequest() {
    }

    public static RefundRequestIntegrationCommand fromBookingRefused(BookingResultEvents.BookingRefused event) {
        Require.requireArgument(event, "event");
        return new RefundRequestIntegrationCommand(
            event.aggregateId().value().toString(),
            event.amount()
        );
    }
}

File: typescript/src/booking/application/integration/payment/adapter/refundRequest.ts
// Anti-Corruption Layer: traduce gli eventi di dominio booking nei command di integrazione del BC payment.

import type { BookingRefused } from '@/booking/domain/events/bookingResultEvents.js';
import { refundRequestIntegrationCommand } from '@/payment/integration/refundRequestIntegrationCommand.js';
import type { RefundRequestIntegrationCommand } from '@/payment/integration/refundRequestIntegrationCommand.js';
import { requireArgument } from '@/common/utils/requireArgument.js';

export function refundRequestFromBookingRefused(
  event: BookingRefused
): RefundRequestIntegrationCommand {
  requireArgument(event, 'event');
  return refundRequestIntegrationCommand(
    event.aggregateId.value.value,
    event.amount
  );
}
